File: exercises/05_exercise_while_loops/02_exam_preparation.py
number_poor_grades = int(input())

still_preparing = True
counter_poor_grades = 0
counter_all_grades = 0
last_problem = " "
sum_grades = 0

while counter_poor_grades < number_poor_grades:
    task_name = input()
    if task_name == "Enough":
        still_preparing = False
        break
    grade = int(input())
    if grade <= 4:
        still_preparing = True
        counter_poor_grades += 1
    sum_grades += grade
    counter_all_grades += 1
    last_problem = task_name

average = sum_grades / counter_all_grades

if not still_preparing:
    print(f"Average score: {average:.2f}")
    print(f"Number of problems: {counter_all_grades}")
    print(f"Last problem: {last_problem}")
else:
    print(f"You need a break, {counter_poor_grades} poor grades.")

File: 05_JavaScriptBasics/02_Exercises/05_whileLoop/02_examPreparation.js
function examPreparation(input){
    index = 0;
    let numBadMarks = Number(input[index]);
    index++;
    let taskName = input[index];
    index++;
    let currentMark = Number(input[index]);
    index++;
    let sumMarks = 0;
    let numTasks = 0;
    let poorMarks = 0;
    let lastTask = "";
    let needBrake = false;

    while (taskName !== "Enough"){
        numTasks += 1;
        lastTask = taskName;
        sumMarks += currentMark;
        
        if (currentMark <= 4){
            poorMarks += 1;
        } 
        if (poorMarks === numBadMarks){
            needBrake = true;
            break;
        }
        
        taskName = input[index];
        index++;
        currentMark = Number(input[index]);
        index++;
    }
    let averageScore = (sumMarks / numTasks).toFixed(2);
    if (needBrake === true){
        console.log(`You need a break, ${poorMarks} poor grades.`);
    } else {
        console.log(`Average score: ${averageScore}`);
        console.log(`Number of problems: ${numTasks}`);
        console.log(`Last problem: ${lastTask}`);
    }
}

examPreparation(["3", "Money", "6", "Story", "4", "Spring Time", "5", "Bus", "6", "Enough"]);
examPreparation(["2", "Income", "3", "Game Info", "6", "Best Player", "4"]);
